File: đồ án thực tế CNPM/Server/src/main/java/com/example/hatd/model/dto/TripRequestResponse.java
package com.example.hatd.model.dto;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

/**
 * DTO (Data Transfer Object): Lớp này CHỈ dùng để chứa thông tin
 * trả về cho Frontend sau khi xử lý yêu cầu.
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class TripRequestResponse {
    
    private Long requestId;       // ID của chuyến đi vừa tạo
    private String status;        // Trạng thái (VD: PENDING)
    private LocalDateTime createdAt;  // Thời gian tạo
    private String message;       // Thông điệp (VD: Tạo thành công)
    
}

File: đồ án thực tế CNPM/Client/app/src/main/java/com/example/myhatd/data/model/TripRequestResponse.kt
package com.example.myhatd.data.model

import com.google.gson.annotations.SerializedName

/**
 * Data Class ánh xạ phản hồi JSON từ Server sau khi gửi yêu cầu chuyến đi thành công.
 * Đây là DTO được sử dụng trong Retrofit API Interface.
 */
data class TripRequestResponse(

    // ID của yêu cầu chuyến đi (requestId)
    @SerializedName("requestId")
    val requestId: Long,

    // Trạng thái của yêu cầu (status) - ví dụ: "PENDING"
    @SerializedName("status")
    val status: String,

    // Thông báo cho người dùng
    @SerializedName("message")
    val message: String,

    // Thời gian yêu cầu được tạo
    // Server Spring Boot sử dụng LocalDateTime, Gson/Retrofit sẽ nhận nó dưới dạng chuỗi
    @SerializedName("createdAt")
    val createdAt: String
)